### PR TITLE
Add config option to allow using Commerce.js client with secret keys

### PR DESCRIPTION
**Pull request by amaau**
Sorry for lines 24 and 25, didn't notice my editor changed it on save.

### CODE REVIEW

**Comment by amaau** [comment]
Addresses [this issue](https://github.com/chec/commerce.js/issues/187).

File: src/commerce.js [hunk 1]
```diff
@@ -13,26 +13,28 @@ const defaultEventCallback = e => {
...
-      throw new Error(
-        'Secret key provided. You must use a public key with Commerce.js!',
-      );
+    if (apiKey.toLowerCase().substring(0, 3) === 'sk_') {
+      if (!config.allowSecretKey) {
+        throw new Error(
+          'Secret key provided. You must use a public key with Commerce.js, or explicitly allow secret keys in the config object.',
+        );
+      }
     }
```
Comment (robbieaverill) [inline src/commerce.js:38]:
```suggestion
    if (!config.allowSecretKey && apiKey.toLowerCase().substring(0, 3) === 'sk_') {
      throw new Error(
        'Secret key provided. You must use a public key with Commerce.js, or explicitly allow secret keys in the config object.',
      );
    }
```

File: src/commerce.js [hunk 2]
```diff
@@ -13,26 +13,28 @@ const defaultEventCallback = e => {
...
 
 class Commerce {
-  constructor(publicKey, debug = false, config = {}) {
+  constructor(apiKey, debug = false, config = {}) {
     this.options = {
       version: 'v1',
       url: 'https://api.chec.io/',
       eventCallback: defaultEventCallback,
       disableStorage: false,
       cartLifetime: 30,
```
Comment (robbieaverill) [inline src/commerce.js:22]:
```suggestion
      cartLifetime: 30,
      allowSecretKey: false,
```

Providing a default value for the config flag

**Review by robbieaverill — COMMENTED** [review]
@ScopeyNZ any concerns with changing the config option name from publicKey to apiKey?

**Comment by amaau** [comment]
@robbieaverill both your edits make sense, but I'm unexperienced with PRs - don't know it it's me who should "commit suggestion" or not

**Comment by robbieaverill** [comment]
@amaau no problem, either of us can commit those suggestions. If you do it through the [changes view](https://github.com/chec/commerce.js/pull/188/files) then you can add them to a batch and commit once. I can also squash the commits on merge to keep it tidy.

**Comment by robbieaverill** [comment]
Thank you for the pull request @amaau 🙂

**Pull request merged**